src/arrays.js

```diff
@@ -1,61 +1,73 @@
 const getNthElement = (index, array) => {
-  // your code here
+  
+  if (index >= array.length){
+    return array[index-array.length];
+  }
+  return array[index];
 };
 
 const arrayToCSVString = array => {
-  // your code here
+const csv = array.toString();
+return(csv);
 };
 
 const csvStringToArray = string => {
-  // your code here
+const array = string.split(',');
+return(array);
 };
 
 const addToArray = (element, array) => {
-  // your code here
+array.push(element);
 };
 
 const addToArray2 = (element, array) => {
-  // your code here
+return array.concat(element);
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+array.splice(index, 1);
 };
 
 const numbersToStrings = numbers => {
-  // your code here
+return numbers.map(String);
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+
 };
 
 const onlyEven = numbers => {
-  // your code here
+  const even = numbers.filter((num) => num % 2 === 0);
+  return even;
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+return array.splice(index, 2,0);
 };
```

Comment:
the problem with this solution is that `splice` mutates the original array and for this test you are required to not alter the original array. To tackle this problem using `splice` you would first need to clone the array. One of the ways of doing this would be to use `map`. e.g.
```
const newArray = array.map(item => item);
```
this would create a copy of the array you could then work with. To remove 1 element at the index your splice would then look like.
```
newArray.splice(index, 1);
```
note: I only want to remove 1 element so my second parameter is `1`. Furthermore, I have removed the `0` from the end of the splice, because any value from the third parameter onwards is added into the array, so if the 0 is left in it would be added into the array at the position you just removed an element.

You can then return this array, overall this would look like:
```
const removeNthElement2 = (index, array) => {
  const newArray = array.map(item => item);
  newArray.splice(index, 1);
  return newArray;
};
```
note: I have returned the `newArray` after doing the `splice`. This is because if I did `return newArray.splice(index, 1)` it would return the removed section from the array not the altered array itself

Reply:
A easier solution to this problem would be to use `filter`, i.e.
```
const removeNthElement2 = (index, array) => {
  return array.filter((_, i) => i !== index);
};
```
I have used `_` for the first parameter in the callback function passed to filter to show that it isn't used. I then used the second parameter, which is the current index in the array, and I check if that index doesn't match the index I want to remove. If it doesn't I add it to the new array otherwise it is filtered out.